### DISCUSSION TITLE
Spring boot integration skywalking agent exception

### DISCUSSION THREAD

**Discussion by zhoucs0504** [apache/skywalking#9591] [Q&A] [answered]
**Environmental dependence：**
jsk11、springboot2.6.7、skywalking9.1.0、skywalking-agent8.12

**Question content:** 
This is a spring cloud project, It is normal for the gateway to join the skywalking agent, However, there are problems with other services.

`ERROR 2022-09-09 16:02:33.031 main SkyWalkingAgent : Enhance class org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor error. 
org.apache.skywalking.apm.agent.core.plugin.PluginException: Can't create InstanceConstructorInterceptorV2.`

Detailed logs are as follows:

[skywalking-api.log](https://github.com/apache/skywalking/files/9533655/skywalking-api.log)

I hope you can help me see what's going on. Thank you in advance

**Comment by wu-sheng** [comment, marked as answer]
Usually this is caused by you added agent jar into the classpath. If not, we are not aware of this.

**Reply by zhoucs0504** [reply]
Thank you for your answer. I have already added the agent package at startup

-javaagent:/data/skywalking-agent/skywalking-agent.jar   -Dskywalking.collector.backend_service=127.0.0.1:11800 -Dskywalking.agent.service_name=myservice-name

**Reply by zhoucs0504** [reply]
I solved it temporarily because the Maven slimming package (spring boot thin layout) was referenced in the project.
If I use Maven slimming package, how can I add skywalking agent to the project?

**Reply by wu-sheng** [reply]
You can't. Agent is not designed for that.

**Reply by wu-sheng** [reply]
Don't add any skywalking package into your maven dependencies tree, except for toolkits.

**Reply by zhoucs0504** [reply]
OK, thank you. I'll revise it
